app/src/main/java/com/google/samples/apps/nowinandroid/ui/NiaAppState.kt

```diff
@@ -85,24 +86,34 @@ class NiaAppState(
     networkMonitor: NetworkMonitor,
     userNewsResourceRepository: UserNewsResourceRepository,
 ) {
-    val currentDestination: NavDestination?
+    val currentBackStackEntry: NavBackStackEntry?
         @Composable get() = navController
-            .currentBackStackEntryAsState().value?.destination
+            .currentBackStackEntryAsState().value
 
     val currentTopLevelDestination: TopLevelDestination?
-        @Composable get() = when (currentDestination?.route) {
-            forYouNavigationRoute -> FOR_YOU
-            bookmarksRoute -> BOOKMARKS
-            interestsRoute -> INTERESTS
-            else -> null
+        @Composable get() {
+            val route: String? = currentBackStackEntry?.destination?.route
+            val arguments: Bundle? = currentBackStackEntry?.arguments
+            return when {
+                route == forYouNavigationRoute -> FOR_YOU
+                route == bookmarksRoute -> BOOKMARKS
+                route == interestsRoute &&
+                    (arguments?.getString(topicIdArg) == null || shouldShowTwoPane) -> INTERESTS
```

Comment:
Those are the conditions that control the visibility of the topbar. Topbar should remain hidden with single pane and topic selected. I will rename this property to  `topBarTitle` or similar as I would like to avoid unnecessary comments.